src/v2/guide/reactivity.md

```diff
@@ -4,80 +4,80 @@ type: guide
 
-## Change Detection Caveats
+## Hạn chế của việc phát hiện thay đổi
 
-Due to the limitations of modern JavaScript (and the abandonment of `Object.observe`), Vue **cannot detect property addition or deletion**. Since Vue performs the getter/setter conversion process during instance initialization, a property must be present in the `data` object in order for Vue to convert it and make it reactive. For example:
+Do những hạn chế của JavaScript hiện tại (và việc `Object.observe` không còn được sử dụng tới), Vue **không thể phát hiện sự thêm hoặc xoá của một thuộc tính nào đó**. Vì Vue thực hiện quá trình chuyển đổi getter / setter trong quá trình khởi tạo thực thể, một thuộc tính bất kì phải được khai báo trong đối tượng `data` để Vue có thể thực hiện chuyển đổi và làm nó trở nên `reactive`. Ví dụ:
 
 ``` js
 var vm = new Vue({
   data: {
     a: 1
   }
 })
-// `vm.a` is now reactive
+// `vm.a` giờ đã trở thành reactive
 
 vm.b = 2
-// `vm.b` is NOT reactive
+// `vm.b` không phải `reactive`
 ```
 
-Vue does not allow dynamically adding new root-level reactive properties to an already created instance. However, it's possible to add reactive properties to a nested object using the `Vue.set(object, key, value)` method:
+Vue không cho phép thêm động các thuộc tính `reactive` ở mức `root` vào một thực thể đã được tạo từ trước. Tuy nhiên, có thể thêm các thuộc tính `reactive` vào một đối tượng lồng nhau bằng cách sử dụng `Vue.set(object, key, value)`:
 
 ``` js
 Vue.set(vm.someObject, 'b', 2)
 ```
 
-You can also use the `vm.$set` instance method, which is an alias to the global `Vue.set`:
+Bạn cũng có thể sử dụng `vm.$set`, đây là một bí danh (alias) toàn cầu của `Vue.set`:
 
 ``` js
 this.$set(this.someObject, 'b', 2)
 ```
 
-Sometimes you may want to assign a number of properties to an existing object, for example using `Object.assign()` or `_.extend()`. However, new properties added to the object will not trigger changes. In such cases, create a fresh object with properties from both the original object and the mixin object:
+Đôi khi bạn có thể muốn gán một số thuộc tính cho một đối tượng hiện có, ví dụ bằng cách sử dụng `Object.assign()` hoặc `_.extend()`. Tuy nhiên, các thuộc tính mới được thêm vào này sẽ không kích hoạt thay đổi (tức là khi chúng thay đổi Vue sẽ không re-render). Trong những trường hợp như thế, bạn nên tạo ra một đối tượng mới với các thuộc tính từ cả đối tượng gốc lẫn đối tượng có chứa các thuộc tính bạn muốn thêm vào, ví dụ:
 
 ``` js
-// instead of `Object.assign(this.someObject, { a: 1, b: 2 })`
+// Không nên `Object.assign(this.someObject, { a: 1, b: 2 })`
 this.someObject = Object.assign({}, this.someObject, { a: 1, b: 2 })
 ```
 
-There are also a few array-related caveats, which were discussed earlier in the [list rendering section](list.html#Caveats).
+Đồng thời cũng vẫn còn tồn tại một số hạn chế liên quan đến Array, những điều đó chúng ta sẽ cùng thảo luộn ở [Render danh sách](list.html#Caveats).
```

Comment:
* "Array" ở đây hoàn toàn có thể dịch thành "mảng."
* "thảo luận" chứ không phải "thảo luộn"
* ý gốc là "_đã_ thảo luận," không phải là "_sẽ_ thảo luận"
* Câu này đọc lên nghe khá lủng củng Có thể dịch ngắn mà chính xác hơn là: Còn một số hạn chế khác liên quan đến mảng mà chúng ta đã thảo luận trong phần "Render danh sách."